validation-test/compiler_crashers_2_fixed/rdar58941114.swift

```diff
@@ -0,0 +1,6 @@
+// RUN: not %target-swift-frontend %s -typecheck
+
+class C {}
+protocol Foo {
+  associatedtype X where C: X
```

Comment:
It would be better if we had a test for this in `tests/` to check the diagnostic message makes sense

Reply:
(I see it makes sense: `qq.swift:3:28: error: type 'C' constrained to non-protocol, non-class type 'Self.X'` but it would be good to ensure it doesn't regress and surface error types or something)